src/swarm-tui/Swarm/TUI/Model/Event.hs

```diff
@@ -0,0 +1,223 @@
+-- ----------------------------------------------
+
+-- | REPL abstract keybinding events.
+--
+-- See 'Swarm.TUI.Controller.REPLEventHandler'.
+data REPLEvent
+  = CancelRunningProgramEvent
+  | TogglePilotingModeEvent
+  | ToggleCustomKeyHandlingEvent
+  deriving (Eq, Ord, Show, Enum, Bounded)
+
+replEvents :: KeyEvents REPLEvent
+replEvents = allKeyEvents $ \case
+  CancelRunningProgramEvent -> "cancel running program"
+  ToggleCustomKeyHandlingEvent -> "toggle custom key handling"
+  TogglePilotingModeEvent -> "toggle piloting mode"
+
+defaultReplBindings :: [(REPLEvent, [Binding])]
+defaultReplBindings = allBindings $ \case
+  CancelRunningProgramEvent -> [ctrl 'c', bind V.KEsc]
+  TogglePilotingModeEvent -> [meta 'p']
+  ToggleCustomKeyHandlingEvent -> [meta 'k']
+
+-- ----------------------------------------------
+--                 REPL EVENTS
+-- ----------------------------------------------
+
+data WorldEvent
+  = ViewBaseEvent
+  | ShowFpsEvent
+  | MoveViewNorthEvent
+  | MoveViewEastEvent
+  | MoveViewSouthEvent
+  | MoveViewWestEvent
+  deriving (Eq, Ord, Show, Enum, Bounded)
+
+worldPanelEvents :: KeyEvents WorldEvent
+worldPanelEvents = allKeyEvents $ \case
+  ViewBaseEvent -> "view base"
+  ShowFpsEvent -> "show fps"
+  MoveViewNorthEvent -> "move view north"
+  MoveViewEastEvent -> "move view east"
+  MoveViewSouthEvent -> "move view south"
+  MoveViewWestEvent -> "move view west"
+
+defaultWorldPanelBindings :: [(WorldEvent, [Binding])]
+defaultWorldPanelBindings = allBindings $ \case
+  ViewBaseEvent -> [bind 'c']
+  ShowFpsEvent -> [bind 'f']
+  MoveViewWestEvent -> [bind 'h', bind V.KLeft]
+  MoveViewSouthEvent -> [bind 'j', bind V.KDown]
+  MoveViewNorthEvent -> [bind 'k', bind V.KUp]
+  MoveViewEastEvent -> [bind 'l', bind V.KRight]
+
+-- ----------------------------------------------
+--                 ROBOT EVENTS
+-- ----------------------------------------------
+
+data RobotEvent
+  = MakeEntityEvent
+  | ShowZeroInventoryEntitiesEvent
+  | CycleInventorySortEvent
+  | SwitchInventorySortDirection
+  | SearchInventoryEvent
+  deriving (Eq, Ord, Show, Enum, Bounded)
+
+robotPanelEvents :: KeyEvents RobotEvent
+robotPanelEvents = allKeyEvents $ \case
+  MakeEntityEvent -> "make entity"
+  ShowZeroInventoryEntitiesEvent -> "show zero inventory entites"
+  CycleInventorySortEvent -> "cycle inventory sort"
+  SwitchInventorySortDirection -> "switch inventory direction"
+  SearchInventoryEvent -> "search inventory"
+
+defaultRobotPanelBindings :: [(RobotEvent, [Binding])]
+defaultRobotPanelBindings = allBindings $ \case
+  MakeEntityEvent -> [bind 'm']
+  ShowZeroInventoryEntitiesEvent -> [bind '0']
+  CycleInventorySortEvent -> [bind ';']
+  SwitchInventorySortDirection -> [bind ':']
+  SearchInventoryEvent -> [bind '/']
+
+-- ----------------
+-- Helper methods
+
+allKeyEvents :: (Ord e, Bounded e, Enum e) => (e -> Text) -> KeyEvents e
+allKeyEvents f = keyEvents $ map (\e -> (f e, e)) [minBound .. maxBound]
```

Comment:
For `[minBound .. maxBound]` we have `listEnums` from `Swarm.Util`.

Also `\e -> (f e, e)` is equivalent to `f &&& id` if you import `Control.Arrow ((&&&))`.  I personally find `f &&& id` clearer than `\e -> (f e, e)` but YMMV.